packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -332,7 +332,7 @@ class ScaffoldState extends State<Scaffold> {
         bottomSheet,
         completer,
         () => entry.remove(),
-        setState
+        (VoidCallback fn) { bottomSheetKey.currentState?.setState(fn); }
```

Comment:
this is fine. I'd even add an assert that bottomSheetKey.currentState isn't null.